### DISCUSSION TITLE
How to revalidate on demand when using to nextjs!

### DISCUSSION THREAD

**Discussion by siamahnaf** [supabase/supabase#19544] [Questions]
Hello, I am using supabase for building a nextjs simple app. I am using app router. In page.tsx I create server component and besically using to fetch supabase data-

`page.tsx`-
```
export const revalidate = 0

//Components
import Lists from "@/Components/Clients/Lists";

//Supabase
import { createServerComponentClient } from "@supabase/auth-helpers-nextjs";
import { cookies } from "next/headers";
import { ClientData } from "@/Supabase/Types/Client/client.types";

const Page = async () => {
    //Client
    const supabase = createServerComponentClient({ cookies });

    //Supabase
    let { data } = await supabase
        .from("client")
        .select("*")
        .order("created_at", { ascending: false })
        .returns<ClientData[]>()

    return (
        <div>
            <Lists data={data} />
        </div>
    );
};

export default Page;
```

So it is fetching data successfully!

Now I have a `List` component that I called into this `page.tsx`

```
"use client"
import { supabase } from "@/Supabase/src/client";

const Lists = ({ data }: Props) => {
    //On Delete Confirm
    const onDeleteConfirm = async(id: string) => {
        const {data, error}=await supabase
        .from("client")
        .delete()
        .eq("id", id)
    }

    return (
        <div className="py-5">
           ......
        </div>
    );
};

export default Lists;
```

And it is `client.ts`-

```
import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";

export const supabase = createClientComponentClient();
```

Here in list component I am deleting data. It's successfully deleting data but I need refresh the cache for updating the list. If you don't understand see the short video please-

[List deleted options __SSDDDD.webm](https://github.com/supabase/supabase/assets/78312010/888c1239-8ba8-4245-8615-151dd808b5ab)

So, how can update the list without browser refreshing. Is there any way to do it using supabase client?

**Comment by truexor** [comment]
### Use `revalidatePath`

Hi, I faced the same issue and got the solution to it. 

By using the [`revalidatePath`](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#revalidating-data) of Nextjs in the server function (down below), the cache will get purged each time 

```
import { revalidatePath } from "next/cache"         //... added

// ....

const Page = async () => {
    //Client
    // ...
    //Supabase
    // ...

    //revalidate path for next return time, replace /page with file route within your app router 
   revalidatePath('/page')

    return (
        // ...
    );
};

export default Page;
```

@siamahnaf
